pkg/testpackage/testpackage.go

```diff
@@ -41,6 +50,18 @@ func NewAnalyzer() *analysis.Analyzer {
 
 				if strings.HasSuffix(fileName, "_test.go") {
 					packageName := f.Name.Name
+
+					allowedPackage := false
+					for _, p := range allowedPackages {
+						if p == packageName {
+							allowedPackage = true
+						}
+					}
+
+					if allowedPackage {
+						continue
+					}
+
 					if !strings.HasSuffix(packageName, "_test") {
 						pass.Reportf(f.Name.Pos(), "package should be `%s_test` instead of `%s`", packageName, packageName)
 					}
```

Comment:
Let's reduce code complexity by extracting function
```suggestion
					processTestFile(pass, f, allowedPackages)
```
```go
func processTestFile(pass *analysis.Pass, f *ast.File, allowedPackages []string) {
	packageName := f.Name.Name

	for _, p := range allowedPackages {
		if p == packageName {
			return
		}
	}

	if !strings.HasSuffix(packageName, "_test") {
		pass.Reportf(f.Name.Pos(), "package should be `%s_test` instead of `%s`", packageName, packageName)
	}
}
```